Source/Libraries/openXDA.Nodes/Types/Analysis/AnalysisNode.cs

```diff
@@ -306,19 +312,51 @@ private void Process(MeterDataSet meterDataSet)
                     }
 
                     Log.Debug($"Finished executing data operation '{dataOperationDefinition.UnqualifiedTypeName}'.");
+                    if (index == 0) status = FileGroupProcessingStatus.Success;
+                    else if (status == FileGroupProcessingStatus.Failed) status = FileGroupProcessingStatus.PartialSuccess;
                 }
                 catch (Exception ex)
                 {
                     // Log the error and skip to the next data operation
                     string message = $"An error occurred while executing data operation of type '{dataOperationDefinition.TypeName}' on data from meter '{meterDataSet.Meter.AssetKey}': {ex.Message}";
                     Exception wrapper = new Exception(message, ex);
+                    if (status == FileGroupProcessingStatus.Success) status = FileGroupProcessingStatus.PartialSuccess;
                     Log.Error(wrapper.Message, wrapper);
+                    dataOperationFailures.Add(new DataOperationFailure
+                    {
+                        DataOperationID = dataOperationDefinition.ID,
+                        FileGroupID = meterDataSet.FileGroup.ID,
+                        Log = message,
+                        StackTrace = ex.StackTrace,
+                        TimeOfFailure = DateTime.UtcNow
+                    });
+                    if (IsDisposed)
+                    {
+                        LogDataOperationFailures(dataOperationFailures);
+                        if (index == 0) return FileGroupProcessingStatus.Failed;
+                        return status;
+                    }
                 }
+
+                index++;
             }
 
-            FileGroup fileGroup = meterDataSet.FileGroup;
+            LogDataOperationFailures(dataOperationFailures);
             _ = NotifyEventEmailNode(fileGroup.ID, fileGroup.ProcessingVersion);
             _ = NotifyEPRICapBankAnalysisNode(fileGroup.ID, fileGroup.ProcessingVersion);
+
+            return status;
+        }
+
+        private void LogDataOperationFailures(IEnumerable<DataOperationFailure> failures)
+        {
+            using (AdoDataConnection connection = CreateDbConnection())
+            {
+                TableOperations<DataOperationFailure> failureTable = new TableOperations<DataOperationFailure>(connection);
+                foreach (DataOperationFailure failure in failures)
+                    failureTable.AddNewRecord(failure);
+            }
+
         }
```

Comment:
We need to make sure if that table is missing it does not take everything down with it.
Probably should tunr this into `TryLogDataOperationFailures` and wrap everything in a `try/catch` with a standard log if it fails.

Reply:
This is a good suggestion, thank you.